src/coreclr/jit/importer.cpp

```diff
@@ -4782,6 +4788,415 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
+            // cgt.un
+            // ret
+
+            GenTree* op2 = impPopStack().val;
+            GenTree* op1 = impPopStack().val;
+
+            GenTree* tmp = gtNewOperNode(GT_GT, TYP_INT, op1, op2);
+            tmp->gtFlags |= GTF_RELOP_NAN_UN | GTF_UNSIGNED;
+            return gtFoldExpr(tmp);
+        }
+
+        case NI_SRCS_UNSAFE_IsAddressLessThan:
+        {
+            assert(sig->sigInst.methInstCount == 1);
+
+            // ldarg.0
+            // ldarg.1
+            // clt.un
+            // ret
+
+            GenTree* op2 = impPopStack().val;
+            GenTree* op1 = impPopStack().val;
+
+            GenTree* tmp = gtNewOperNode(GT_LT, TYP_INT, op1, op2);
+            tmp->gtFlags |= GTF_RELOP_NAN_UN | GTF_UNSIGNED;
+            return gtFoldExpr(tmp);
+        }
+
+        case NI_SRCS_UNSAFE_IsNullRef:
+        {
+            assert(sig->sigInst.methInstCount == 1);
+
+            // ldarg.0
+            // ldc.i4.0
+            // conv.u
+            // ceq
+            // ret
+            CLANG_FORMAT_COMMENT_ANCHOR;
+
+#ifdef TARGET_64BIT
+            bool uns = true;
+#else
+            bool uns                    = false;
+#endif
+
+            GenTree* op1 = impPopStack().val;
+
+            GenTree* cns = gtNewIconNode(0, TYP_INT);
+
+#ifdef TARGET_64BIT
+            cns = gtNewCastNode(TYP_I_IMPL, cns, uns, TYP_U_IMPL);
+#endif
+
+            GenTree* tmp = gtNewOperNode(GT_EQ, TYP_INT, op1, cns);
+            return gtFoldExpr(tmp);
+        }
+
+        case NI_SRCS_UNSAFE_NullRef:
+        {
+            assert(sig->sigInst.methInstCount == 1);
+
+            // ldc.i4.0
+            // conv.u
+            // ret
+            CLANG_FORMAT_COMMENT_ANCHOR;
+
+#ifdef TARGET_64BIT
+            bool uns = true;
+#else
+            bool uns                    = false;
+#endif
+
+            GenTree* cns = gtNewIconNode(0, TYP_INT);
+
+#ifdef TARGET_64BIT
+            cns = gtNewCastNode(TYP_I_IMPL, cns, uns, TYP_U_IMPL);
+#endif
+
+            return cns;
+        }
+
+        case NI_SRCS_UNSAFE_Read:
+        {
+            assert(sig->sigInst.methInstCount == 1);
+
+            // ldarg.0
+            // ldobj !!T
+            // ret
+
+            return nullptr;
+        }
+
+        case NI_SRCS_UNSAFE_ReadUnaligned:
+        {
+            assert(sig->sigInst.methInstCount == 1);
+
+            // ldarg.0
+            // unaligned. 0x1
+            // ldobj !!T
+            // ret
+
+            return nullptr;
+        }
+
+        case NI_SRCS_UNSAFE_SizeOf:
+        {
+            assert(sig->sigInst.methInstCount == 1);
+
+            // sizeof !!T
+            // ret
+
+            unsigned classSize = info.compCompHnd->getClassSize(sig->sigInst.methInst[0]);
+            return gtNewIconNode(classSize, TYP_INT);
+        }
+
+        case NI_SRCS_UNSAFE_SkipInit:
+        {
+            assert(sig->sigInst.methInstCount == 1);
+
+            // ret
+
+            impPopStack();
```

Comment:
This needs to handle the case when the value has side effects.

Something like `if ((op1->gtFlags & GTF_SIDE_EFFECT) != 0)  { op1 = gtUnusedVal(op1); }` should work well enough I think.

Reply:
Fixed.